modules/hal_ambiq/Kconfig.components

```
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2025 Antmicro <www.antmicro.com>
+
+menu "Bluetooth"
+	visible if SOC_BT_SUPPORTED
+
+	config BT_ENABLED
+		bool "Bluetooth"
+		depends on SOC_BT_SUPPORTED
+		help
+		  Select this option to enable Bluetooth and show the submenu with Bluetooth configuration choices.
```

Seems risky to define a `CONFIG_BT_` value in a module rather in the Zephyr subsystem. 

Maybe 
```suggestion
	config BT_AMBIQ_ENABLED
		bool "Bluetooth"
		depends on SOC_BT_SUPPORTED
		help
		  Select this option to enable Bluetooth and show the submenu with Bluetooth configuration choices.
```
is less risky

---

Sure, thanks.

---

Same with `SOC_BT_SUPPORTED`. That's not a generic option, rather than an ambiq specific one, so it should have "AMBIQ" somewhere in the name. That said, can't all of this be done with device tree, i.e. "SoC supports BT" is the same as there being an HCI DT node for the ambiq HCI driver? And why do you need `BT_ENABLED`? Why not just use the existing `BT`? Seems very odd to have platform specific options like this for something that really shouldn't be platform specific.

---

Ok, changed to SOC_AMBIQ_BT_SUPPORTED. 
Adding `SOC_AMBIQ_BT_SUPPORTED` will prevent someone tries to add an HCI node to not supported SoCs either on application level or board level.
The `AMBIQ_BT_ENABLED` is changed to `AMBIQ_COMPONENT_USE_BT` to align with our `AMBIQ_HAL_USE_*` kconfigs.

---

> such as BT_CUSTOM

Out of curiosity, do you actually use this? We haven't had any known users of `BT_CUSTOM` since the early days of Zephyr, and certainly no upstream users of it, so I've entertained the thought of just removing this.

---

> > such as BT_CUSTOM
> 
> Out of curiosity, do you actually use this? We haven't had any known users of `BT_CUSTOM` since the early days of Zephyr, and certainly no upstream users of it, so I've entertained the thought of just removing this.

No, I don't think so, just trying to complete the logic here.